dotnet/src/Connectors/Connectors.Memory.Sqlite/SqliteGenericDataModelMapper.cs

```diff
@@ -0,0 +1,146 @@
+    IVectorStoreRecordMapper<VectorStoreGenericDataModel<ulong>, Dictionary<string, object?>>,
+    IVectorStoreRecordMapper<VectorStoreGenericDataModel<string>, Dictionary<string, object?>>
+{
+    /// <summary><see cref="VectorStoreRecordPropertyReader"/> with helpers for reading vector store model properties and their attributes.</summary>
+    private readonly VectorStoreRecordPropertyReader _propertyReader;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SqliteGenericDataModelMapper"/> class.
+    /// </summary>
+    /// <param name="propertyReader">A <see cref="VectorStoreRecordDefinition"/> that defines the schema of the data in the database.</param>
+    public SqliteGenericDataModelMapper(VectorStoreRecordPropertyReader propertyReader)
+    {
+        Verify.NotNull(propertyReader);
+
+        this._propertyReader = propertyReader;
+    }
+
+    #region Implementation of IVectorStoreRecordMapper<VectorStoreGenericDataModel<string>, Dictionary<string, object?>>
+
+    public Dictionary<string, object?> MapFromDataToStorageModel(VectorStoreGenericDataModel<string> dataModel)
+    {
+        return this.InternalMapFromDataToStorageModel(dataModel);
+    }
+
+    public VectorStoreGenericDataModel<string> MapFromStorageToDataModel(Dictionary<string, object?> storageModel, StorageToDataModelMapperOptions options)
+    {
+        return this.InternalMapFromStorageToDataModel<string>(storageModel, options);
+    }
+
+    #endregion
+
+    #region Implementation of IVectorStoreRecordMapper<VectorStoreGenericDataModel<ulong>, Dictionary<string, object?>>
+
+    public Dictionary<string, object?> MapFromDataToStorageModel(VectorStoreGenericDataModel<ulong> dataModel)
+    {
+        return this.InternalMapFromDataToStorageModel(dataModel);
+    }
+
+    VectorStoreGenericDataModel<ulong> IVectorStoreRecordMapper<VectorStoreGenericDataModel<ulong>, Dictionary<string, object?>>.MapFromStorageToDataModel(Dictionary<string, object?> storageModel, StorageToDataModelMapperOptions options)
+    {
+        return this.InternalMapFromStorageToDataModel<ulong>(storageModel, options);
+    }
+
+    #endregion
+
+    #region private
+
+    private Dictionary<string, object?> InternalMapFromDataToStorageModel<TKey>(VectorStoreGenericDataModel<TKey> dataModel)
+        where TKey : notnull
+    {
+        var properties = new Dictionary<string, object?>
+        {
+            // Add key property
+            { this._propertyReader.KeyPropertyStoragePropertyName, dataModel.Key }
+        };
+
+        // Add data properties
+        if (dataModel.Data is not null)
+        {
+            foreach (var property in this._propertyReader.DataProperties)
+            {
+                if (dataModel.Data.TryGetValue(property.DataModelPropertyName, out var dataValue))
+                {
+                    properties.Add(this._propertyReader.StoragePropertyNamesMap[property.DataModelPropertyName], dataValue);
```

Comment:
You can also use propertyReader.GetStoragePropertyName for storage lookups where you have the propertyreader available.